src/main/java/org/springframework/integration/dsl/transaction/TransactionInterceptorFactoryBean.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.interceptor.DefaultTransactionAttribute;
+import org.springframework.transaction.interceptor.MatchAlwaysTransactionAttributeSource;
+import org.springframework.transaction.interceptor.TransactionAttribute;
+import org.springframework.transaction.interceptor.TransactionInterceptor;
+
+/**
+ * Provides a fluent API to build a transaction interceptor. See
+ * {@link TransactionAttribute} for property meanings; if a {@link TransactionAttribute}
+ * is provided, the individual properties are ignored. If a
+ * {@link PlatformTransactionManager} is not provided, we look for a bean with name
+ * "transactionManager".
+ * @author Gary Russell
+ * @since 1.2
+ *
+ */
+public class TransactionInterceptorFactoryBean extends AbstractFactoryBean<TransactionInterceptor> {
+
+	private Propagation propagation = Propagation.REQUIRED;
+
+	private Isolation isolation = Isolation.DEFAULT;
+
+	private int timeout = -1;
+
+	private boolean readOnly = false;
+
+	private TransactionAttribute transactionAttribute;
+
+	private PlatformTransactionManager transactionManager;
+
+	public TransactionInterceptorFactoryBean propagation(Propagation propagation) {
+		this.propagation = propagation;
+		return this;
+	}
+
+	public TransactionInterceptorFactoryBean isolation(Isolation isolation) {
+		this.isolation = isolation;
+		return this;
+	}
+
+	public TransactionInterceptorFactoryBean timeout(int timeout) {
+		this.timeout = timeout;
+		return this;
+	}
+
+	public TransactionInterceptorFactoryBean readOnly(boolean readOnly) {
+		this.readOnly = readOnly;
+		return this;
+	}
+
+	public TransactionInterceptorFactoryBean transactionAttribute(TransactionAttribute transactionAttribute) {
+		this.transactionAttribute = transactionAttribute;
+		return this;
+	}
+
+	public TransactionInterceptorFactoryBean transactionManager(PlatformTransactionManager transactionManager) {
+		this.transactionManager = transactionManager;
+		return this;
+	}
+
+	@Override
+	protected TransactionInterceptor createInstance() throws Exception {
+		MatchAlwaysTransactionAttributeSource attSource = new MatchAlwaysTransactionAttributeSource();
+		if (this.transactionAttribute == null) {
+			DefaultTransactionAttribute transactionAttribute = new DefaultTransactionAttribute();
+			transactionAttribute.setPropagationBehavior(this.propagation.value());
+			transactionAttribute.setIsolationLevel(this.isolation.value());
+			transactionAttribute.setTimeout(this.timeout);
+			transactionAttribute.setReadOnly(this.readOnly);
+			attSource.setTransactionAttribute(transactionAttribute);
+		}
+		else {
+			attSource.setTransactionAttribute(this.transactionAttribute);
+		}
+		if (this.transactionManager == null) {
+			this.transactionManager = this.getBeanFactory().getBean("transactionManager",
+					PlatformTransactionManager.class);
```

Comment:
I'm intended to disagree. See `TransactionAspectSupport.determineTransactionManager()`:
```
else if (StringUtils.hasText(this.transactionManagerBeanName)) {
	return determineQualifiedTransactionManager(this.transactionManagerBeanName);
}
else {
	PlatformTransactionManager defaultTransactionManager = getTransactionManager();
	if (defaultTransactionManager == null) {
		defaultTransactionManager = this.beanFactory.getBean(PlatformTransactionManager.class);
		this.transactionManagerCache.putIfAbsent(
		DEFAULT_TRANSACTION_MANAGER_KEY, defaultTransactionManager);
	}
	return defaultTransactionManager;
}
```
I don't say that we should allow to provide `transactionManagerBeanName`, but at the same time we shouldn't be so strict for the `defaultTransactionManager`. 
In other words let's rely on the `TransactionAspectSupport` logic and don't do any `getBean()` here!

Therefore this class can be freed from the `AbstractFactoryBean` oppression, and may just turn to the beautiful butterfly - raw Builder class :smile: .

Good work anyway!